locator/src/negative_cache.rs

```diff
@@ -25,12 +26,12 @@ impl NegativeCache {
 
     pub fn contains(&self, key: &str) -> bool {
         let cache_hit = self.cache.contains_key(key);
-        let metric_name = if cache_hit {
-            "negative_cache.hit"
+        let metric_def = if cache_hit {
+            NEGATIVE_CACHE_HIT
         } else {
-            "negative_cache.miss"
+            NEGATIVE_CACHE_MISS
         };
-        metrics::counter!(metric_name).increment(1);
+        metrics::counter!(metric_def.name).increment(1);
```

Comment:
Should we have a lint check to ensure that all metric names defined in as a `MetricDef`?